File: Angular/project/hello-ang/src/app/employee/Employee.ts
export class Employee {
    ID: number;
    Name: string;
    FirstName: string = "Hadeer";
    LastName: string;
    Age: number;
    Mobile: string;
    Email: string;

    constructor(name?: string, lastName?: string, email?: string, age?: number, mobile?: string) {
        if (this.FirstName === undefined) {
            this.FirstName = "Hadeer"
        }
        else
            this.FirstName = name;
        if (this.Email === undefined) {
            this.Email = "Elnaggar"
        }
        else
            this.Email = email;
        if (age === undefined) {
            this.Age = 20;
        }
        else
            this.Age = age;
        if (mobile === undefined) {
            this.Mobile = "01206968970";
        }
        else
            this.Mobile = mobile
        if (this.LastName === undefined) {
            this.LastName = "elnaggar"
        }
        else
            this.LastName = lastName
        this.Name = this.getFullName();    
    }


    getFullName(): string {
        return this.FirstName + " " + this.LastName;
    }

    setFirstName(name: string) {
        this.FirstName = name;
    }
}

File: Restful WebServices/RESTful webservices/src/main/java/iti/hadeer/models/Employee.java
package iti.hadeer.models;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.NamedQuery;
import javax.persistence.Table;

import jakarta.xml.bind.annotation.XmlRootElement;


@XmlRootElement
@Entity
@Table(name = "employee")

@NamedQuery(
    name = "selectAll",
    query = "select e from Employee e"
)
public class Employee{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int id;

    @Column(name = "firstname")
    String firstName;

    @Column
    String email;

    @Column(name = "lastname")
    String lastName;

    @Column
    int age;

    public Employee(){

    }

    public Employee(String firstName, String lastName, String email, int age){
        this.age=age;
        this.email=email;
        this.firstName=firstName;
        this.lastName=lastName;
    }

    public void setFirstName(String firstName){
        this.firstName=firstName;
    }

    public String getFirstName(){
        return firstName;
    }

    public void setLastName(String lastName){
        this.lastName=lastName;
    }

    public String getLastName(){
        return lastName;
    }

    public void setEmail(String email){
        this.email=email;
    }

    public String getEmail(){
        return email;
    }

    public void setAge(int age){
        this.age=age;
    }

    public int getAge(){
        return age;
    }


}